### PR TITLE
use goderive to generate deepcopy code

### PR DESCRIPTION
**Pull request by ndeloof**
This replaces copystructyre (based on reflection) with `deepcopy` code generated by [goderive](https://github.com/awalterschulze/goderive)

compose project loading pipeline being a mess, we mutate the project in a few places and need a full copy of it. This has terrible impact when it comes to large compose project:

```
$ docker compose version
Docker Compose version v2.27.1

$ time docker compose ps nginx23
NAME             IMAGE          COMMAND                  SERVICE   CREATED       STATUS                PORTS
toti-nginx23-1   nginx:1.22.1   "/docker-entrypoint.…"   nginx23   7 hours ago   Up 7 hours (Paused)   80/tcp

real	0m2.638s
user	0m3.170s
sys	0m0.149s


# With this PR
$ docker compose version
Docker Compose version v2.27.1-4-g02f40eea6.m

$ time docker compose ps nginx23
NAME             IMAGE          COMMAND                  SERVICE   CREATED       STATUS                PORTS
toti-nginx23-1   nginx:1.22.1   "/docker-entrypoint.…"   nginx23   7 hours ago   Up 7 hours (Paused)   80/tcp

real	0m0.905s
user	0m0.210s
sys	0m0.082s
```

I had to [fork](https://github.com/awalterschulze/goderive/pull/87) the original project so `Extensions` can define a custom `DeepCopy` method and we are not stuck my `interface{}` not being a supported type

fixes https://github.com/docker/compose/issues/11881

### CODE REVIEW

**Review by glours — APPROVED** [review]

File: types/derived.gen.go [hunk 1]
```diff
@@ -0,0 +1,2067 @@
...
+func deriveDeepCopy_55(dst, src *DiscreteGenericResource) {
+	dst.Kind = src.Kind
+	dst.Value = src.Value
+	if src.Extensions != nil {
+		dst.Extensions = make(map[string]any, len(src.Extensions))
+		src.Extensions.DeepCopy(dst.Extensions)
+	} else {
+		dst.Extensions = nil
+	}
+}
```
Comment (jhrotko) [inline types/derived.gen.go:2067]: damn

**Review by jhrotko — APPROVED** [review]
LGTM

File: types/derived.gen.go (hunk 1 above)
Comment (ndeloof) [reply types/derived.gen.go:2067]: I'm happy I didn't had to write this by hand :)

**Pull request merged**